### PR TITLE
Use clevercsv

### PR DESCRIPTION
**Pull request by lisad**
I'm rather tempted to stop here, rather than go further down the path of using the builtin csv reader and building 
our own dicts. 

Pros: 
* clevercsv supports stream dict reading, so that would allow a pipeline using that to operate row-by-row and handle large data sets pretty easily some day when we do the other work
* I like the overall approach of being RFC 4180 strict
* The problem with header row containing spaces and not stripping quotes is manageable in phaser code since we're canonicalizing column names anyway
* Empty lines are handled EXCEPT if it's the first line - seems reasonable
* I figured out a way to error if there are duplicate column names - clevercsv has a table reader that just lists all the column names rather than turn them into a dict (and lose the info about dupes)
* Figured out a way to identify rows with too many commas

Cons:
* Comment lines are not handle (and they're not RFC4180 compatible either)
* Rows with not enough values fill in the last values with None -- similar to pandas.read_csv --  not as much warning generation as we would be able to do if we wrote our own
 
At this point I think I can live with the tradeoffs WDYT @jeffkole

### CODE REVIEW

File: phaser/phase.py [hunk 1]
```diff
@@ -93,8 +93,8 @@ def df_transform(self, df_data):
         raise PhaserError("Subclass DataFramePhase and return a new dataframe in the 'df_transform' method")
 
     def load_data(self, data):
-        # Overrides the regular load_data because we just want to accept dataframe and keep it in df format.
-        self.df_data = data
+        # Overrides the regular load_data because we want to convert incoming data to DataFrame.
+        self.df_data = pd.DataFrame(data)
```
Comment (jeffkole) [inline phaser/phase.py:97]: This is much better than my solution to this problem had been.

File: phaser/util.py
Comment (jeffkole) [inline phaser/util.py:1]: I'm thinking that this file might be better off being named `io.py` and contain all of the functions we will want for managing io.  I have a bit of an allergy to modules named `util` is they are not really descriptive of their purpose.

File: phaser/util.py [hunk 2]
```diff
@@ -1,22 +1,11 @@
...
-    * Raises errors loading 'bad lines', rather than skip
-    """
-    return pd.read_csv(source,
-                       dtype='str',
-                       sep=',',
-                       skip_blank_lines=True,
-                       index_col=False,
-                       comment='#')
+    stream = stream_table(source)
+    header = next(stream)
```
Comment (jeffkole) [inline phaser/util.py:7]: Without reading clevercsv's documentation or code, I'm assuming that `stream_table(source)` followed by `next(stream)` only reads the first line and does not read the entire file.  Is that correct?  (A reasonable answer is "RTFM.")

File: tests/test_csv.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+
+@pytest.mark.skip("Commented lines are really hard to skip with clevercsv but otherwise it's a good library...")
+def test_comment_lines(tmpdir):
+    with open(tmpdir / 'commented_line', 'w') as f:
+        f.write("#crew\n,id,name\n,1,James Kirk\n")
+    assert read_csv(tmpdir / 'commented_line') == [{'id':1, 'name':'James Kirk'}]
+
+
+def test_empty_lines_at_end(tmpdir):
```
Comment (jeffkole) [inline tests/test_csv.py:47]: All of these test functions write to the same file, don't they?

When we have a huge test corpus and want to run tests in parallel, I'm guessing this is going to get difficult to debug.

^ Passive aggressive way of saying, "Let's give these files distinct names."

**Review by jeffkole — APPROVED** [review]

File: phaser/util.py
Comment (lisad) [reply phaser/util.py:1]: that's a great idea, I'll do that next.  i do have a habit of factoring stuff out of 'util' once I know what they are, so we're pretty much on the same page anyway.

File: phaser/util.py (hunk 2 above)
Comment (lisad) [reply phaser/util.py:7]: Yes you're right and I'll leave a comment here to help future readers not have to RTFM or guess.

File: tests/test_csv.py (hunk 3 above)
Comment (lisad) [reply tests/test_csv.py:47]: No - the magic 'tmpdir' fixture is supposed to create a different tmpdir for every test.  it's a nice feature of pytest.

**Pull request merged**